analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/SonarDiagnosticAnalyzer.cs

```diff
@@ -36,7 +37,21 @@ public sealed override void Initialize(RoslynAnalysisContext context)
             {
                 context.EnableConcurrentExecution();
             }
-            Initialize(new SonarAnalysisContext(context, SupportedDiagnostics));
+
+            if (ShouldRegisterAnalyzerPredicate != null)
+            {
+                context.RegisterCompilationStartAction(compilation =>
```

Comment:
Will this work for actions that register their own CompilationStartAction inside this CompilationStartAction? Just out of curiosity